src/main/java/org/zalando/pazuzu/sort/TopologicalSortLinear.java

```diff
@@ -0,0 +1,129 @@
+package org.zalando.pazuzu.sort;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * Created by dimangaliyev on 25/04/16.
+ * Topological sort algorithm
+ * Running time complexity O(n + m)
+ * where n - number of vertices, m - number of edges
+ */
+public class TopologicalSortLinear<T> {
+
+    /**
+     * Color enum for vertex
+     * WHITE - not visited
+     * GREY - visited in process of traversal
+     * BLACK - visited
+     */
+    private enum Color {
+        WHITE, GREY, BLACK
+    }
+
+    // List of all vertices
+    private Collection<T> vertices;
+
+    // Function for returning all children of vertex
+    private Function<T, Set<T>> getChildren;
+
+    // Visited indicator for vertex
+    private Map<T, Color> color;
+
+    // Parent of vertex
+    // key - vertex, value - parent of vertex
+    private Map<T, T> parent;
+
+    public TopologicalSortLinear(Collection<T> vertices, Function<T, Set<T>> getChildren) {
+        this.vertices = vertices;
+        this.getChildren = getChildren;
+
+        this.color = new HashMap<>(vertices.size());
+        // mark all vertices as not visited
+        for(T v : vertices) color.put(v, Color.WHITE);
```

Comment:
Please stick to common code style in the project:

for (...) {
  // code...
}